File: VecMathREPLCpp/main/Scalar.cpp
#include "Scalar.h"

Scalar::Scalar() :Scalar(0)
{
}

Scalar::Scalar(float value) :IMatrix(
	Type::VECTOR,
	AddressMode::CLAMP,
	1,
	1
), m_Value(value)
{
}

float Scalar::get(int ri, int ci)
{
	if (inRange(ri, ci)) {
	return m_Value;
	}
	else {
		switch (m_AddressMode) {
		case AddressMode::CLAMP:return m_Value;
		case AddressMode::BORDER:return m_BorderValue;
		case AddressMode::BOUNDSERROR:return NAN;
		}
	}
}

void Scalar::set(int ri, int ci,float value)
{
	if (inRange(ri, ci)) {
		m_Value = value;
	}
}

File: VecMathREPLJava/src/main/java/dae/gd/vecmathproject/Scalar.java
package dae.gd.vecmathproject;

/**
 *
 * @author Koen.Samyn
 */
public class Scalar extends IMatrix {

    private float value;

    public Scalar() {
        this(0);
    }

    public Scalar(float x) {
        super(Type.VECTOR, AddressMode.CLAMP, 1, 1);
        this.value = x;
    }

    @Override
    public float get(int ri, int ci) {
        if (inRange(ri, ci)) {
            return value;
        } else {
            switch (addressMode) {
                case CLAMP:
                    return value;
                case BORDER:
                    return borderValue;
                case BOUNDSERROR:
                    return Float.NaN;
                default:
                    return Float.NaN;
            }
        }
    }

    @Override
    public void set(int ri, int ci, float value) {
        if (ri == 0 && ci == 0) {
            this.value = value;
        }
    }
}
